File: 240422/겹치지 않는 선분 2/line-segments-that-do-not-overlap-2.py
N = int(input())
lines = [list(map(int, input().split())) for _ in range(N)]

def check_cross(num):
    some = 0

    p0 = lines[num][0]
    p1 = lines[num][1]
    p_diff = p1 - p0

    for p in range(N):
        if p == num:
            continue

        cp0, cp1 = lines[p][0], lines[p][1]
        cp_diff = cp1 - cp0

        up = p0 * cp_diff - cp0 * p_diff
        down = cp_diff - p_diff

        if down != 0:
            result = up / down

            if min(p0, p1) <= result <= max(p0, p1) and min(cp0, cp1) <= result <= max(cp0, cp1):
                some += 1
    
    if some == 0:
        return True
    else:
        return False

cnts = 0
for i in range(N):
    if check_cross(i):
        cnts += 1

print(cnts)

# # 모범답안
# # 변수 선언 및 입력
# n = int(input())
# x = [
#     tuple(map(int, input().split()))
#     for _ in range(n)
# ]

# ans = 0


# # 다른 선분과 겹치지 않는 선분의 수를 구합니다.
# for i in range(n):
# 	# i번째 선분이 다른 선분과 겹치지 않는지 확인합니다.
# 	overlap = False
	
# 	for j in range(n):
# 		# 자기 자신은 제외합니다.
# 		if j == i:
# 			continue
		
# 		# x1이 큰 쪽이 x2가 더 작다면 겹치게 됩니다.
# 		if (x[i][0] <= x[j][0] and x[i][1] >= x[j][1]) or (x[i][0] >= x[j][0] and x[i][1] <= x[j][1]):
# 			overlap = True
# 			break
	
#     # 겹치지 않았다면 정답의 개수에 하나를 추가합니다.
# 	if overlap == False:
# 		ans += 1

# print(ans)

File: 260305/겹치지 않는 선분 2/line-segments-that-do-not-overlap-2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N;
    cin >> N;

    vector<pair<long long, long long>> lines(N);
    for (int i = 0; i < N; i++) 
    {
        long long a, b;
        cin >> a >> b;
        lines[i] = {a, b};
    }

    // Please write your code here.
    auto check_cross = [&](int num) -> bool 
    {
        int some = 0;

        long long p0 = lines[num].first;
        long long p1 = lines[num].second;
        long long p_diff = p1 - p0;

        for (int p = 0; p < N; p++) 
        {
            if (p == num) continue;

            long long cp0 = lines[p].first;
            long long cp1 = lines[p].second;
            long long cp_diff = cp1 - cp0;

            long long up = p0 * cp_diff - cp0 * p_diff;
            long long down = cp_diff - p_diff;

            if (down != 0) 
            {
                long double result = (long double)up / (long double)down;

                long double pmin = (long double)min(p0, p1);
                long double pmax = (long double)max(p0, p1);
                long double cpmin = (long double)min(cp0, cp1);
                long double cpmax = (long double)max(cp0, cp1);

                if (pmin <= result && result <= pmax &&
                    cpmin <= result && result <= cpmax) 
                {
                    some += 1;
                }
            }
        }

        return (some == 0);
    };

    int cnts = 0;
    for (int i = 0; i < N; i++) {
        if (check_cross(i)) cnts++;
    }

    cout << cnts << "\n";
    return 0;
}